hnn_core/tests/test_network.py

```diff
@@ -96,16 +110,32 @@ def test_spikes():
     with pytest.raises(ValueError, match="No input types found for ABC"):
         spikes.plot_hist(spike_types='ABC', show=False)
 
+    with pytest.raises(ValueError, match="Invalid mean_type. Valid "
+                       "arguments include 'all', 'trial', or 'cell'."):
+        spikes.mean_rates(mean_type='ABC')
```

Comment:
can we have more thorough tests for `mean_rates`? Like have some simulated spikes and check that the spike rate comes out to be what you expected. Try different corner cases, e.g., when one trial has no spikes -- does the mean come out correctly etc.

Reply:
Looks like the manually written `spikes` object in `test_network.py` has this exact corner case (2 trials, each cell only spikes once per trial). I've added tests asserting the correct calculation for all three `mean_types`. I'll try and think of trickier edge cases before merging but this seems sufficient for now.